ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/storage/BonsaiWorldStateKeyValueStorage.java

```diff
@@ -203,6 +204,10 @@ public Optional<byte[]> getTrieLog(final Hash blockHash) {
     return trieLogStorage.get(blockHash.toArrayUnsafe());
   }
 
+  public Stream<byte[]> streamTrieLogs() {
+    return trieLogStorage.streamKeys();
```

Comment:
Used to preload the pruner. Need to test the feasibility of this on real nodes, might need to limit it.

Reply:
yes seems to be dangerous to do that without limit

Reply:
agreed. If we do not close the RocksIterator, that is going to lead to problems.  I would suggest passing in a Consumer<TrieLog> or something like that so we can compartmentalize the iterator open/close

Reply:
Now I'm simply using `streamKeys().limit(..)`, hopefully that is safe!

[`8658aec` (#6026)](https://github.com/hyperledger/besu/pull/6026/commits/8658aec93fbe1d54d0166247062808f53dbc8629)